tools/data-handler/src/containers/project.ts

```diff
@@ -342,7 +374,9 @@ export class Project extends CardContainer {
     from: ResourcesFrom = ResourcesFrom.localOnly,
   ): Promise<Resource[]> {
     const moduleCalculations =
-      await this.collectResourcesFromModules('calculations');
+      from !== ResourcesFrom.localOnly
```

Comment:
The local resources won't be updated right? Now that constructor is only ran once, should we move `resourcesSync`-calls from constructor to these methods. Same issue with all other methods

Reply:
True. I'd like to avoid scanning the disc for all of the calls for resources (e.g. call to `templates()`) 
There's the `addResource()` and `removeResource()` methods now. Preferably we should call "sync" only when there is a plethora of new potential resources; or we don't know the state.

Reply:
I moved the resources collection to happen in a method `collectResources()`. It is now called from `Project` constructor and after renaming happens. Are there any other places that would require it?

Reply:
Maybe not part of this PR, but would it be easier if we did all file system level logic inside Project? Now it's up to the user of project to remember to notify the project if there are changes to the state of the Project

Reply:
Yes, it would be better to have file system notification logic. That way user cannot "sneak in" changes. 
We might have some old ticket about this. NodeJS is probably not the greatest platform for it, but we'll use what we have.